cpp/arcticdb/version/local_versioned_engine.cpp

```diff
@@ -1054,10 +1054,33 @@ std::vector<std::pair<VersionedItem, FrameAndDescriptor>> LocalVersionedEngine::
     return folly::collect(results_fut).get();
 }
 
-std::vector<AtomKey> LocalVersionedEngine::batch_write_internal(
+folly::Future<VersionedItem> LocalVersionedEngine::async_write_index_key_to_version_map(
+    const std::shared_ptr<VersionMap> &version_map,
+    const AtomKey&& index_key,
+    const UpdateInfo& stream_update_info,
+    bool prune_previous_versions) {
+
+    folly::Future<folly::Unit> write_version_fut;
+
+    if(prune_previous_versions) {
+        write_version_fut = async::submit_io_task(WriteAndPrunePreviousTask{store(), version_map, index_key, stream_update_info.previous_index_key_});
+    } else {
+        write_version_fut = async::submit_io_task(WriteVersionTask{store(), version_map, index_key});
+    }
+
+    return std::move(write_version_fut)
+    .then([this, index_key, store = store()](auto &&){
```

Comment:
`index_key` can be captured by reference, and there's no need to capture `store` explicitly when you have `this`